apps/dashboard/test/integration/apps_test.rb

```diff
@@ -0,0 +1,43 @@
+require 'test_helper'
+
+class AppsTest < ActionDispatch::IntegrationTest
```

Comment:
As an aside to this PR, this is a new Integration test because [rails docs on controller unit tests](https://api.rubyonrails.org/v6.1.0/classes/ActionController/TestCase.html) discourage controller unit tests and say "Functional tests allow you to test a single controller action per test method." which we have been severely overloading. (I ran into this issue trying to stub my way through resetting controller state).

So probably moving forward, we'd want to migrate our controller tests to integration tests and generally be wary of PRs with unit tests against controllers.

Reply:
I'm mostly okay with that. In certain cases I used controller tests mainly because it was easier to write at the time and I was more familiar with it (coming from Rails 3).

There are definitely quite a few places where the controller tests could be removed by moving logic into model code, and then adding unit tests instead.